File: cpp/011-020/Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) {
            return "";
        }
        string prefix = strs[0];
        
        for (int i = 1; i < strs.size(); i++) {
            const string& str = strs[i];
            
            int k = 0;
            while (k < min(prefix.length(), str.length())) {
                if (prefix[k] != str[k]) {
                    break;
                }
                k++;
            }
            
            prefix = prefix.substr(0, k);
        }
        
        return prefix;
    }
};

File: scala/0011-0020/Longest Common Prefix.scala
import scala.util.control.Breaks._

object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
        if (strs.isEmpty) {
            return ""
        }
        
        var prefix = strs(0)
        
        for (i <- 1 until strs.length) {
            val str = strs(i)
            
            var k = 0;
            
            breakable {
                while (k < math.min(prefix.length, str.length)) {
                    if (prefix(k) != str(k)) { 
                        break
                    }
                    k += 1
                }
            }
            
            prefix = prefix.substring(0, k)
        }
        
        return prefix;
    }
}
